### DISCUSSION TITLE
Release 0.5

### DISCUSSION THREAD

**Discussion by N4S4** [N4S4/synology-api#113] [Announcements]
## What's Changed
* Error codes by @eddybl in https://github.com/N4S4/synology-api/pull/108
* Error codes handling by @N4S4 in https://github.com/N4S4/synology-api/pull/109
* Add dict output option for start requests by @fcouziniedevy in https://github.com/N4S4/synology-api/pull/110
* Certificate export method by @Yurzs in https://github.com/N4S4/synology-api/pull/112
* expire_times (download counts) added to filestation edit_shared_link() and create_shared_link()
* Add_dict_output_option_for_start_requests for Filestation and Downloadstation
* alarm_rules_logs(), alarm_rules_list(), resource_monitor_settings_list(), file_handling_access(), list_service_group(), list_process_group(), installed_package_list(), active_notifications() functions added to core_sys_info.py 
* Login error handling improved
* Adapting code to convention
* **Breaking**: dhcp_server.py Class dhcp_server is now DhcpServer to follow the class naming convention
* **Breaking**: drive_admin_console.py Class admin_console is now AdminConsole
* **Breaking**: usb_copy.py Class USB_Copy is now USBCopy

## New Contributors
* @fortysix2ahead for the initial error_code handling
* @eddybl made their first contribution in https://github.com/N4S4/synology-api/pull/108
* @fcouziniedevy made their first contribution in https://github.com/N4S4/synology-api/pull/110
* @Yurzs made their first contribution in https://github.com/N4S4/synology-api/pull/112

**Full Changelog**: https://github.com/N4S4/synology-api/compare/v0.4.4...v0.5

<hr /><em>This discussion was created from the release <a href='https://github.com/N4S4/synology-api/releases/tag/v0.5'>Release 0.5</a>.</em>

**Comment by WesleyHilhorst** [comment]
Hi! Do you by any chance know how to implement the encrypted authentication correctly? The DS apps seem to be using `SYNO.API.Encryption` to call 'SYNO.API.Auth' afterwards.

**Reply by N4S4** [reply]
I did not investigate further encryption Api, this is because you encountered some issues?

**Reply by WesleyHilhorst** [reply]
Did not encounter issues. But would want to provide maximum safety :-).

**Reply by N4S4** [reply]
True, well don't know much about `SYNO.API.Encryption` I will investigate more and see differences between both methods. If you have more info or docs let me know.

**Reply by WesleyHilhorst** [reply]
What I know is that the public key should be used (in combination with key/token) to encrypt the auth `SYNO.API.Auth` parameters. But how that should be done is not documented anywhere exactly.

**Reply by N4S4** [reply]
Yeah the very little documentation is the problem, been asking Synology support for some api info or advanced docs but no response at all, I will see what I find! Btw thank you for letting me know!

**Reply by WesleyHilhorst** [reply]
I did find some code on Github that shows some form of encrypted authentication, I personally couldn't get it to work with so far: 

https://github.com/openstack/cinder/blob/61d506880eafcfcfee9047ac182a45d555e32a22/cinder/volume/drivers/synology/synology_common.py

https://github.com/Coollision/synology-videostation-index-updater/blob/e2360025f15c2a100579b12c741a41590ea16221/synology/session/login.go

**Reply by N4S4** [reply]
Well I see that a bit to digest, I had no idea about this. 
Interesting and i will search more about this topic and move this conversation to a new branch (If I'll understand something)

**Reply by WesleyHilhorst** [reply]
Cool, looking forward to it!
